Sources/SnapshotPreviewsCore/UIKitRenderingStrategy.swift

```diff
@@ -33,38 +33,51 @@ public class UIKitRenderingStrategy: RenderingStrategy {
 
   @MainActor
   public func render(
-    preview: SnapshotPreviewsCore.Preview,
-    completion: @escaping (SnapshotResult) -> Void
+      preview: SnapshotPreviewsCore.Preview,
+      completion: @escaping (SnapshotResult) -> Void
   ) {
-    Self.setup()
-    let previewOrientation = preview.orientation.toInterfaceOrientation()
-    if windowScene!.interfaceOrientation != previewOrientation {
-      var rotationError: (any Error)? = nil
+      Self.setup()
+      let targetOrientation = preview.orientation.toInterfaceOrientation()
+      guard windowScene!.interfaceOrientation != targetOrientation else {
+          performRender(preview: preview, completion: completion)
+          return
+      }
+
       if #available(iOS 16.0, *) {
-        windowScene!.requestGeometryUpdate(.iOS(interfaceOrientations: previewOrientation.toInterfaceOrientationMask())) { error in
-          NSLog("Rotation error handler: \(error) \(self.windowScene!.interfaceOrientation)")
-          rotationError = error
-        }
+          var geometryUpdateError = false
+          windowScene!.requestGeometryUpdate(.iOS(interfaceOrientations: targetOrientation.toInterfaceOrientationMask())) { error in
+              NSLog("Rotation error handler: \(error) \(self.windowScene!.interfaceOrientation)")
+              geometryUpdateError = true
+              completion(SnapshotResult(image: .failure(error), precision: nil, accessibilityEnabled: nil, accessibilityMarkers: nil, colorScheme: nil))
+              return
+          }
+          if !geometryUpdateError {
```

Comment:
This doesn't seem like it will work, the error handler is called async so it won't be set by the time waitForOrientationChange is called - and then that kicks off an async task which will need to check if the error has been set by the time it checks again